runtime/interpreter/value.go

```diff
@@ -19757,8 +19757,14 @@ func DereferenceValue(
 	locationRange LocationRange,
 	referenceValue ReferenceValue,
 ) Value {
-	referencedValue := referenceValue.ReferencedValue(inter, locationRange, true)
-	return (*referencedValue).Transfer(
+	referencedValue := *referenceValue.ReferencedValue(inter, locationRange, true)
+
+	// Defensive check: ensure that the referenced value is not a resource
+	if referencedValue.IsResourceKinded(inter) {
+		panic(errors.NewUnreachableError())
```

Comment:
Maybe throw a dedicated error

Reply:
Good point, improved in 34e9e43d1e6b3ad0a4bf34b26a621cdb097360a5